drivers/retained_mem/retained_mem_zephyr_ram.c

```diff
@@ -124,6 +132,7 @@ static const struct retained_mem_driver_api zephyr_retained_mem_ram_api = {
 			zephyr_retained_mem_ram_config_##inst = {				\
 		.address = (uint8_t *)DT_REG_ADDR(DT_PARENT(DT_INST(inst, DT_DRV_COMPAT))),	\
 		.size = DT_REG_SIZE(DT_PARENT(DT_INST(inst, DT_DRV_COMPAT))),			\
+		RETAINED_MEM_MUTEX(inst)							\
```

Comment:
@nordicjm Which value is set for `config->mutex` in `reatined_mem_zephyr_ram.c` if no DT node is defined for retained mem driver, i.e. just the retention DT node is defined? I'm asking that because by reading the code I would assume it would be `true` however I'm getting it set to `false`:

```
#0  zephyr_retained_mem_ram_lock_take (dev=dev@entry=0x800deb8 <__device_dts_ord_154>) at /home/gromero/zephyrproject/zephyr/drivers/retained_mem/retained_mem_zephyr_ram.c:33
#1  0x08007300 in zephyr_retained_mem_ram_read (dev=0x800deb8 <__device_dts_ord_154>, offset=0, buffer=0x20007f58 <z_interrupt_stacks+1928> '\252' <repeats 16 times>, "\f", size=2)
    at /home/gromero/zephyrproject/zephyr/drivers/retained_mem/retained_mem_zephyr_ram.c:83
#2  0x080040a8 in z_impl_retained_mem_read (dev=dev@entry=0x800deb8 <__device_dts_ord_154>, offset=offset@entry=0, buffer=buffer@entry=0x20007f58 <z_interrupt_stacks+1928> '\252' <repeats 16 times>, "\f", size=size@entry=2)
    at /home/gromero/zephyrproject/zephyr/include/zephyr/drivers/retained_mem.h:132
#3  0x080040ea in retained_mem_read (dev=0x800deb8 <__device_dts_ord_154>, offset=0, buffer=buffer@entry=0x20007f58 <z_interrupt_stacks+1928> '\252' <repeats 16 times>, "\f", size=size@entry=2)
    at /home/gromero/zephyrproject/zephyr/build/PR59264/instrumentation/zephyr/include/generated/syscalls/retained_mem.h:59
#4  0x080043c2 in retention_is_valid (dev=0x800ded0 <__device_dts_ord_155>) at /home/gromero/zephyrproject/zephyr/subsys/retention/retention.c:188
#5  0x08001c14 in instr_init () at /home/gromero/zephyrproject/zephyr/subsys/instrumentation/common/instr_common.c:94
#6  0x0800dc90 in __cyg_profile_func_enter (callee=callee@entry=0x8009ac5 <z_early_memcpy>, caller=caller@entry=0x800d451 <z_data_copy+36>) at /home/gromero/zephyrproject/zephyr/subsys/instrumentation/handlers/instr_handlers_gcc.c:25
#7  0x08009adc in z_early_memcpy (dst=0x20000000 <_char_out>, src=0x800e9a8, n=0) at /home/gromero/zephyrproject/zephyr/kernel/init.c:131
#8  0x0800d450 in z_data_copy () at /home/gromero/zephyrproject/zephyr/kernel/xip.c:27
#9  0x08002e42 in z_arm_prep_c () at /home/gromero/zephyrproject/zephyr/arch/arm/core/aarch32/prep_c.c:264
#10 0x08003860 in z_arm_reset () at /home/gromero/zephyrproject/zephyr/arch/arm/core/aarch32/cortex_m/reset.S:169
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
(gdb) n
38		if (config->mutex) {
(gdb) list
33	{
34	#ifdef CONFIG_RETAINED_MEM_MUTEXES
35		struct zephyr_retained_mem_ram_data *data = dev->data;
36		const struct zephyr_retained_mem_ram_config *config = dev->config;
37	
38		if (config->mutex) {
39			k_mutex_lock(&data->lock, K_FOREVER);
40		}
41	#else
42		ARG_UNUSED(dev);
(gdb) p config->mutex
$1 = false
(gdb) 
```
It means that now "by default" the mutexes in retained mem driver are disabled, i.e. if I don't force any mutex disabled by CONFIG or in the DT node I get mutexes disabled. Is that behavior intentional -- it's different from the current one, where mutexes are enabled by default?

I think that happen because no DT node is defined for retention mem? But on the other way it's a bit tricky to make config->mutex in retained mem diver dependent on the no-mutex attribute in retention DT node?

Reply:
It seems that the DT macro I used does not actually work on bool entries, which is not documented, so have switched to an alternative macro which works.

Reply:
Got it. Now the default behavior is ok (true). However I'm still not able to disable the mutexes by only adding attribute `no-mutex`  to the retention DT node. Just to be sure I'm not missing something. Is the idea that the following overlay would disable mutexed for both retention and the retained mem driver?

```
/*
 * Copyright 2023 Linaro
 *
 * SPDX-License-Identifier: Apache-2.0
 */

#include <common/mem.h>

/ {
        sram@200BF000 {
                compatible = "zephyr,memory-region", "mmio-sram";
                reg = <0x200BF000 0x20>;
                zephyr,memory-region = "RetainedMem";
                status = "okay";

                retainedmem {
                        compatible = "zephyr,retained-ram";
                        status = "okay";
                        #address-cells = <1>;
                        #size-cells = <1>;

                        retention0: retention@0 {
                                compatible = "zephyr,retention";
                                status = "okay";

                                no-mutex;

                                reg = <0x0 0x20>;

                                prefix = [BE EF];
                        };
               };
        };
};

&sram0 {
       reg = <0x20000000 DT_SIZE_K(764)>;
};
```

or  it's missing that attribute in another DT node?

Reply:
It also needs it in the retained memory part too, so would be listed in the overlay twice

Reply:
@nordicjm Awesome, all good so! I've just tested and it works nicely. So think it's only missing to address the build issue with `bool` missing a string (my last review).

Reply:
Just for the records and others, that's an example for disabling the mutexes using DT attribute `no-mutex`:

```
/*
 * Copyright 2023 Linaro
 *
 * SPDX-License-Identifier: Apache-2.0
 */

#include <common/mem.h>

/ {
        sram@200BF000 {
                compatible = "zephyr,memory-region", "mmio-sram";
                reg = <0x200BF000 0x20>;
                zephyr,memory-region = "RetainedMem";
                status = "okay";

                retainedmem {
                        compatible = "zephyr,retained-ram";
                        status = "okay";
                        #address-cells = <1>;
                        #size-cells = <1>;

                        no-mutex;

                        retention0: retention@0 {
                                compatible = "zephyr,retention";
                                status = "okay";

                                reg = <0x0 0x20>;

                                no-mutex;

                                prefix = [BE EF];
                        };
               };
        };
};

&sram0 {
       reg = <0x20000000 DT_SIZE_K(764)>;
};
```